packages/devtools_app/lib/src/service/editor/api_classes.dart

```diff
@@ -256,6 +289,196 @@ class ThemeChangedEvent extends EditorEvent {
   Map<String, Object?> toJson() => {Field.theme: theme};
 }
 
+/// An event sent by an editor when the current cursor position/s change.
+class ActiveLocationChangedEvent extends EditorEvent {
+  ActiveLocationChangedEvent({
+    required this.selections,
+    required this.textDocument,
+  });
+
+  ActiveLocationChangedEvent.fromJson(Map<String, Object?> map)
+    : this(
+        textDocument: TextDocument.fromJson(
+          map[Field.textDocument] as Map<String, Object?>,
+        ),
+        selections:
+            (map[Field.selections] as List<Object?>)
+                .cast<Map<String, Object?>>()
+                .map(EditorSelection.fromJson)
+                .toList(),
+      );
+
+  final List<EditorSelection> selections;
+  final TextDocument textDocument;
+
+  @override
+  EditorEventKind get kind => EditorEventKind.activeLocationChanged;
+
+  @override
+  Map<String, Object?> toJson() => {
+    Field.selections: selections,
+    Field.textDocument: textDocument,
+  };
+}
+
+/// A reference to a text document in the editor.
+class TextDocument with Serializable {
+  TextDocument({required this.uri, required this.version});
+
+  TextDocument.fromJson(Map<String, Object?> map)
+    : this(uri: map[Field.uri] as String, version: map[Field.version] as int);
+
+  final String uri;
```

Comment:
nit: let's name this `uriAsString` to be consistent with other parts of the code that store a uri as a string value. Can you also add dartdoc describing what type of URI this is? e.g. file uri, absolute, etc

Reply:
done!